addons/purchase_mrp/models/stock_move.py

```diff
@@ -20,6 +20,8 @@ def _get_price_unit(self):
             return super()._get_price_unit()
         line = self.purchase_line_id
         kit_price_unit = line._get_gross_price_unit()
+        if line.currency_id != self.company_id.currency_id:
+            kit_price_unit = line.currency_id._convert(kit_price_unit, self.company_id.currency_id, self.company_id, fields.Date.context_today(self), round=False)
```

Comment:
For the `fields.Date.context_today(self)`, I followed the same logic as here:
https://github.com/odoo/odoo/blob/5eddeeb64b7394e0c9bc409bb3c74582a13fbb4a/addons/purchase_stock/models/stock_move.py#L76